File: frontend/src/api/BusinessObject.js
export default class BusinessObject {

   constructor() {
        this.name = "";
        this.id = 0;
        this.creation_date = new Date()
   }

    setName(name) {
        this.name = name
    }

    getName() {
        return this.name
    }

    setID(id) {
        this.id = id
    }

    getID() {
        return this.id
    }

    setCreationDate(creation_date) {
        this.creation_date = creation_date
    }

    getCreationDate() {
        return this.creation_date
    }

    toString() {
        let result = ''
        for (var prop in this) {
            result += prop + ': ' + this[prop] + ' ';
        }
        return result;
    }

}

File: src/server/bo/BusinessObject.py
from abc import ABC
from datetime import datetime


class BusinessObject(ABC):
    """
    Definition der Basisklasse aller BusinessObjects in diesem Projekt.

    Jedes BusinessObject hat drei Attribute creation_date, id und name.
    """

    def __init__(self):
        self._creation_date = datetime.now()
        self._id = 0
        self._name = ""

    def get_creation_date(self):
        """Auslesen des Erzeugungsdatum."""
        return self._creation_date

    def set_creation_date(self, date):
        """Setzen des Erzeugungsdatum"""
        self._creation_date = date

    def get_id(self):
        """Auslesen der id."""
        return self._id

    def set_id(self, id):
        """Setzen der id."""
        self._id = id

    def get_name(self):
        """Auslesen der Namen."""
        return self._name

    def set_name(self, name):
        """Setzen des Namens."""
        self._name = name
